File: 1325_delete_leaves_with_a_given_value/removeLeafNodes.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def removeLeafNodes(self, root: Optional[TreeNode], target: int) -> Optional[TreeNode]:
        if root.left:
            left = self.removeLeafNodes(root.left, target)
            root.left = left
        if root.right:    
            right = self.removeLeafNodes(root.right, target)
            root.right = right

        if root.val == target and root.left == root.right:
            return None
        else:
            return root   

        return root

File: 1325_delete_leaves_with_a_given_value/removeLeafNodes.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
struct TreeNode* removeLeafNodes(struct TreeNode* root, int target) {
    if (root -> left) {
        root -> left = removeLeafNodes(root -> left, target);
    }
    if (root -> right) {
        root -> right = removeLeafNodes(root -> right, target);
    }

    if (root -> left == root -> right && root -> val == target) {
        return 0;
    } else {
        return root;
    }
}